src/commands/offenses/getfilteredoffenses.js

```diff
@@ -0,0 +1,53 @@
+const fetch = require("node-fetch")
+const { apiurl } = require("../../../config.json")
+const { MessageEmbed } = require("discord.js")
+const ConfigModel = require("../../database/schemas/config")
+
+
+module.exports = {
+    config: {
+        name: "getfilteredoffenses",
+        aliases: [],
+        usage: "<playername>",
+        category: "offenses",
+        description: "Gets all offenses of a player from only trusted communities",
+        accessibility: "Member",
+    },
+    run: async (client, message, args) => {
+        if (!args[0]) return message.reply("Provide a player name to get offenses of")
+        const playername = args.shift()
+        const offensesRaw = await fetch(`${apiurl}/offenses/getall?playername=${playername}`)
+        const offenses = await offensesRaw.json()
+        if (offenses === null)
+            return message.channel.send(`User \`${playername}\` has no offenses!`)
+
+        const config = await ConfigModel.findOne({ guildid: message.guild.id })
+        if (!config)
+            return message.reply("No server config!")
+        
+        let embed = new MessageEmbed()
+            .setTitle("FAGC Offenses")
+            .setColor("GREEN")
+            .setTimestamp()
+            .setAuthor("FAGC Community")
+            .setDescription(`FAGC Offense of player \`${playername}\``)
+        const communities = await (await fetch(`${apiurl}/communities/getall`)).json()
+        let i = 0
+        offenses.forEach((offense) => {
+            if (i == 25) {
+                message.channel.send(embed)
+                embed.fields = []
+            }
+            communities.forEach((community) => {
+                if (config.trustedCommunities.some(id => {
```

Comment:
Replaced with this instead. Returns, which basically lets it pass to the next iteration.
```js
communities.forEach((community) => {
    if (config.trustedCommunities.find(communityID => community.name === offense.communityname && community._id === communityID)) return
    const violations = offense.violations.map((violation) => { return violation._id })
    embed.addField(offense._id, `Community name: ${offense.communityname}, Violation ID(s): ${violations.join(", ")}`)
    i++
})
```

Reply:
I think this now does the opposite to what you wanted.
If the community matches the offense and is trusted, then it is not added to the embed.
But when the community name does not mattch or it is not trusted, then it is added to the embed.
The amount of time it took me to work out the logic also suggests it is not a good solution.

Reply:
I'm doing two things at the same time, which is probably what made it harder to understand. Took me a while too.
You are also correct about the condition, it is wrong, as that condition was meant for the rest of the code to be inside of it, such as `if find() { //code }`. Changed and added comments. Is this better?
```js
if (config.trustedCommunities.find(communityID => community._id != communityID)) return // if the community is not trusted, go to the next community
if (community.name != offense.communityname) // if the name of the community is not the name of the community in the offense, then go to the next community
```

Reply:
Yeah that will work now, and this would be the only option if communities didnt have unique names.
I could be wrong, but I think the communities have unique names which would make `.find()` more efficient than `.forEach()` even with returns.
Here is what it would look like:
```js
let community = communities.find(community => community.name == offense.communityname)
if (config.trustedCommunities.includes(community._id) {
    // Add violation to embed
}
```